addons/sale_timesheet_margin/models/sale_order_line.py

```diff
@@ -8,9 +8,10 @@ class SaleOrderLine(models.Model):
     @api.depends('analytic_line_ids.amount', 'qty_delivered_method')
     def _compute_purchase_price(self):
         timesheet_sols = self.filtered(
-            lambda sol: sol.qty_delivered_method == 'timesheet' and not sol.product_id.standard_price
+            lambda sol: sol.qty_delivered_method == 'timesheet' and not sol.product_id.standard_price and not sol.product_id.service_policy == 'ordered_timesheet'
         )
-        super(SaleOrderLine, self - timesheet_sols)._compute_purchase_price()
+        already_computed_service = self.filtered(lambda sol: sol.create_date is not False and sol.product_id.service_policy == 'ordered_timesheet')
```

Comment:
When the product is not a service but a consumable one than it could be satisfied in this filtered. I think it is not expected. 🤔 
We could probably check if the SOL contains a service product. 🙂 
cc @IT-Ideas

Reply:
My bad, there is no default value in the service_policy field as II thought, and it seems in 15.0 the cost in SOL containing a consumable product does not change after confirming without this fix. So, LGTM, thanks. 🙂

Reply:
You made me doubt for a moment 😅

And yea to be precise : for consumable product, service_policy is either equals to `False` (when invoice_policy = 'order') or `'delivered_manual'` (when invoice_policy = 'delivery').

Thanks for the review 🙂